### ISSUE TITLE
Panic when using Azure Service Bus queue that doesn't exist

### ISSUE THREAD

**Issue by protonyx** [kedacore/keda#3563]
### Report

When using an Azure Service Bus scaler using a queue that doesn't exist, KEDA enters a crash loop. In the logs, we see the message: "panic: runtime error: invalid memory address or nil pointer dereference"

### Expected Behavior

The scaledObject should be marked as failed or inactive with a message about an invalid subscription

### Actual Behavior

Crash loop in the KEDA operator

### Steps to Reproduce the Problem

1. Create a Service Bus TriggerAuthentication and secret with a Service Bus connection string
2. Create a scaledObject that specifies a queueName that doesn't exist

Example objects:

```
apiVersion: v1
kind: Secret
metadata:
  name: asb-connection
  namespace: keda
data:
  connection: >-
    ***REDACTED***
type: Opaque
---
apiVersion: keda.sh/v1alpha1
kind: TriggerAuthentication
metadata:
  name: asb-trigger-auth
  namespace: keda
spec:
  secretTargetRef:
    - key: connection
      name: asb-connection
      parameter: connection
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  namespace: keda
  labels:
    app: nginx
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.14.2
        ports:
        - containerPort: 80
---
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: bug-test
  namespace: keda
spec:
  cooldownPeriod: 300
  maxReplicaCount: 20
  minReplicaCount: 2
  pollingInterval: 30
  scaleTargetRef:
    name: nginx-deployment
  triggers:
    - authenticationRef:
        name: asb-trigger-auth
      metadata:
        messageCount: '50'
        queueName: invalid-queue
      type: azure-servicebus
```

### Logs from KEDA operator

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x2ae7b29]

goroutine 487 [running]:
github.com/kedacore/keda/v2/pkg/scalers.getQueueEntityFromNamespace({0x3cae988, 0xc00045a340}, 0xc00045a340, {0xc000f020f0, 0x23})
	/workspace/pkg/scalers/azure_servicebus_scaler.go:320 +0x69
github.com/kedacore/keda/v2/pkg/scalers.(*azureServiceBusScaler).getAzureServiceBusLength(0xc000830690, {0x3cae988, 0xc00045a340})
	/workspace/pkg/scalers/azure_servicebus_scaler.go:274 +0xc5
github.com/kedacore/keda/v2/pkg/scalers.(*azureServiceBusScaler).IsActive(0xc000830690, {0x3cae988, 0xc00045a340})
	/workspace/pkg/scalers/azure_servicebus_scaler.go:189 +0x2f
github.com/kedacore/keda/v2/pkg/scaling/cache.(*ScalersCache).IsScaledObjectActive(0xc00040fa40, {0x3cae988, 0xc00045a340}, 0xc000ba1000)
	/workspace/pkg/scaling/cache/scalers_cache.go:89 +0xef
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers(0xc000bc03f0, {0x3cae988, 0xc00045a340}, {0x34e3ec0, 0xc000ba1000}, {0x3c8b6e8, 0xc001650580})
	/workspace/pkg/scaling/scale_handler.go:278 +0x4b2
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop(0xc000bc03f0, {0x3cae988, 0xc00045a340}, 0xc001660140, {0x34e3ec0, 0xc000ba1000}, {0x3c8b6e8, 0xc001650580})
	/workspace/pkg/scaling/scale_handler.go:149 +0x31c
created by github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).HandleScalableObject
	/workspace/pkg/scaling/scale_handler.go:105 +0x6ef
```

### KEDA Version

2.8.0

### Kubernetes Version

1.22

### Platform

Microsoft Azure

### Scaler Details

Azure Service Bus

**Comment by tomkerkhove** [comment]
@v-shenoy Is this something you can quickly check please?

**Comment by v-shenoy** [comment]
I tried to reproduce the issue with the following ScaledObject
```yaml
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: sb-bug
spec:
  scaleTargetRef:
    name: nginx-deployment
  pollingInterval:  10                             
  cooldownPeriod:   30                            
  minReplicaCount:  0                                 
  maxReplicaCount:  5                             
  triggers:
  - type: azure-servicebus
    metadata:
      connectionFromEnv: connection
      topicName: topic-exists
      subscriptionName: sub-doesnot-exist
      messageCount: "5"
      activationMessageCount: "0"
```

The operator logs an error, instead of panicking. 
You mention that you are using a topic/subscription but the stack trace contains a call to `getQueueEntityFromNamespace` which should not happen. 

Could you share your ScaledObject / TriggerAuth (if any) definitions?

**Comment by protonyx** [comment]
You are absolutely right, we were attempting to use a topic/subscription, but the scaledObject was referencing a queueName instead of the topicName/subscriptionName. When referencing a subscription that doesn't exist, there is no error. But I was able to reproduce the issue with an invalid queueName using this set of objects:

```
apiVersion: v1
kind: Secret
metadata:
  name: asb-connection
  namespace: keda
data:
  connection: >-
    ***REDACTED***
type: Opaque
---
apiVersion: keda.sh/v1alpha1
kind: TriggerAuthentication
metadata:
  name: asb-trigger-auth
  namespace: keda
spec:
  secretTargetRef:
    - key: connection
      name: asb-connection
      parameter: connection
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  namespace: keda
  labels:
    app: nginx
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.14.2
        ports:
        - containerPort: 80
---
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: bug-test
  namespace: keda
spec:
  cooldownPeriod: 300
  maxReplicaCount: 20
  minReplicaCount: 2
  pollingInterval: 30
  scaleTargetRef:
    name: nginx-deployment
  triggers:
    - authenticationRef:
        name: asb-trigger-auth
      metadata:
        messageCount: '50'
        queueName: invalid-queue
      type: azure-servicebus
```

**Comment by v-shenoy** [comment]
I have a Service Bus with queues `valid-queue-1` and `valid-queue-2`. I created a SAS Policy for `valid-queue-1` and used the primary connection string in the secret (exact yaml as you provided). In the ScaledObject definition, I provided the `queueName` as `valid-queue-2`. 

Here's the error I get in the operator pods - 
```
2022-08-18T21:51:20Z    ERROR   azure_servicebus_scaler error   {"type": "ScaledObject", "namespace": "keda", "name": "bug-test", "error": "request failed: 401 Unauthorized"}
github.com/kedacore/keda/v2/pkg/scaling/cache.(*ScalersCache).IsScaledObjectActive
        /workspace/pkg/scaling/cache/scalers_cache.go:89
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
        /workspace/pkg/scaling/scale_handler.go:278
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
        /workspace/pkg/scaling/scale_handler.go:149
2022-08-18T21:51:20Z    ERROR   azure_servicebus_scaler error   {"type": "ScaledObject", "namespace": "keda", "name": "bug-test", "error": "request failed: 401 Unauthorized"}
github.com/kedacore/keda/v2/pkg/scaling/cache.(*ScalersCache).IsScaledObjectActive
        /workspace/pkg/scaling/cache/scalers_cache.go:94
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
        /workspace/pkg/scaling/scale_handler.go:278
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
        /workspace/pkg/scaling/scale_handler.go:149
2022-08-18T21:51:20Z    ERROR   scalehandler    Error getting scale decision    {"scaledobject.Name": "bug-test", "scaledObject.Namespace": "keda", "scaleTarget.Name": "nginx-deployment", "error": "request failed: 401 Unauthorized"}
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
        /workspace/pkg/scaling/scale_handler.go:278
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
        /workspace/pkg/scaling/scale_handler.go:149
```

I can't seem to get the operator to panic. Am I missing something here?

**Comment by protonyx** [comment]
I ran the same steps that caused it to panic yesterday and am no longer getting the same output. Now I get the same "401 Unauthorized" error that you posted. I looked back through our scaledObjects and found 7 other misconfigured scaledObjects, but only 1 of them was causing the panic (along with the example I was using). Now that I can't reproduce it, I am not sure how to go about getting more information to determine the cause. The SAS tokens might not have been the true cause.
